src/utils/benefitController.js

```diff
@@ -0,0 +1,74 @@
+import { DECEMBER, BENEFIT_PRICE, MAIN_MENUS, DESSERTS, BENEFIT_MESSAGE, MENUS } from "../constants/index.js";
+import { getCategoryMenuCount } from "./event.js";
+
+const benefitController = () => {
+  const benefitDiscounts = {
+    [BENEFIT_MESSAGE.CHRISTMAS]: 0,
+    [BENEFIT_MESSAGE.WEEKDAYS]: 0,
+    [BENEFIT_MESSAGE.WEEKEND]: 0,
+    [BENEFIT_MESSAGE.SPECIAL]: 0,
+    [BENEFIT_MESSAGE.GIFT]: 0,
+  };
+
+  const setInit = () => {
+    Object.keys(benefitDiscounts).forEach((key) => {
+      benefitDiscounts[key] = 0;
+    });
+  };
+
+  const getBenefitDiscounts = () => {
+    return { ...benefitDiscounts };
+  };
+
+  // 크리스마스 디데이 할인
+  const setChristmasBenefit = (visitDate) => {
+    if (visitDate > DECEMBER.CHRISTMAS) {
+      return;
+    }
+
+    benefitDiscounts[BENEFIT_MESSAGE.CHRISTMAS] = (visitDate - 1) * BENEFIT_PRICE.HUNDRED + BENEFIT_PRICE.THOUSAND;
+  };
+
+  // 평일, 주말 할인
+  const setDayBenefit = (isWeekend, orderMenus) => {
+    if (isWeekend) {
+      const mainMenuCount = getCategoryMenuCount(MAIN_MENUS, orderMenus);
+      benefitDiscounts[BENEFIT_MESSAGE.WEEKEND] = mainMenuCount * BENEFIT_PRICE.YEAR;
+      return;
+    }
+
+    const dessertMenuCount = getCategoryMenuCount(DESSERTS, orderMenus);
+    benefitDiscounts[BENEFIT_MESSAGE.WEEKDAYS] = dessertMenuCount * BENEFIT_PRICE.YEAR;
+  };
```

Comment:
```suggestion
  const setDayBenefit = (isWeekend, orderMenus) => {
    const category = isWeekend ? MAIN_MENUS : DESSERTS;
    const benefit = isWeekend ? BENEFIT_MESSAGE.WEEKEND : BENEFIT_MESSAGE.WEEKDAYS;
    const menuCount = getCategoryMenuCount(category, orderMenus);
    benefitDiscounts[benefitType] = menuCount * BENEFIT_PRICE.YEAR;
  };
```
현재는 가독성 측면에서 보기 좋은 것 같은데 
주말인경우랑 평일이 중복되는 것 같아요 ! 예시 처럼 한번에 처리하는 건 어떠세요 ?!

Reply:
오 좋은 것 같습니다👍